test/unit/ColorConversionTests.js

```diff
@@ -0,0 +1,47 @@
+const {test, Test} = require('tap');
+
+const {rgbToHsv, hsvToRgb} = require('../../src/util/color-conversions');
+
+Test.prototype.addAssert('colorsAlmostEqual', 2, function (found, wanted, message, extra) {
+    /* eslint-disable no-invalid-this */
+    message += `: found ${JSON.stringify(Array.from(found))}, wanted ${JSON.stringify(Array.from(wanted))}`;
+
+    // should always return another assert call, or
+    // this.pass(message) or this.fail(message, extra)
+    if (found.length !== wanted.length) {
+        return this.fail(message, extra);
+    }
+
+    for (let i = 0; i < found.length; i++) {
+        if (Math.abs(found[i] - wanted[i]) > 1e-3) {
```

Comment:
It'd be nice to use a tolerance that means something, rather than just an arbitrary `> 1e-3`. Maybe `>= 0.5 / 360`, so it'll reject any error that would cause variance in a hue value after rounding?